src/reachability/utils/reachability.jl

```diff
@@ -3,12 +3,51 @@
 # Performs layer-by-layer propagation
 # It is called by all solvers under reachability
 # TODO: also called by ReluVal and FastLin, so move to general utils (or to network.jl)
-function forward_network(solver, nnet::Network, input)
-    reach = input
+function forward_network(solver::Solver, nnet::Network, input;
+                         collect=false, before_act=false, overapprox=false, T=Any)
+    if collect
+        f = before_act ? first : last
+        g = overapprox ? overapproximate : identity
+        collect_f = g∘f
+        return _forward_network_collect(solver, nnet, input, collect_f, T)
+    else
+        return _forward_network(solver, nnet, input)
+    end
+end
+
+# these two functions can instead just be exposed without the leading "_",
+function _forward_network(solver::Solver, nnet::Network, input)
+    Z = input
     for layer in nnet.layers
-        reach = forward_layer(solver, layer, reach)
+        Ẑ, Z = forward_layer(solver, layer, Z)
     end
-    return reach
+    return Z
+end
+
+function _forward_network_collect(solver::Solver,
+                                  nnet::Network,
+                                  input,
+                                  collect_f=last,
+                                  T=Any)
+    zs = (input, input)
+    collection::Vector{T} = [collect_f(zs)]
+    for L in nnet.layers
+        zs = forward_layer(solver, L, zs[2])
+        push!(collection, collect_f(zs))
+    end
+    return collection
+end
+function get_bounds(solver, nnet, input; before_act=false)
```

Comment:
Since we're adding a `get_bounds(solver, nnet, input; before_act)`, we should probably change the definition of the original (IA) get_bounds to have the same keyword, rather than optional argument act. Currently it is `get_bounds(nnet, input[, act])`

Reply:
`get_bounds(problem::Problem, args...) = get_bounds(problem.network, problem.input, args...)
`
Is there a good way to handle this line if I make the original get_bounds require a keyword? I'm not too familiar with the args... and how I can make it work

Reply:
You can forward both regular and keyword arguments if that's what you mean. E.g. you can have something like:
```julia
f(x, y; z = 1) = x + y + z
f(x, args...; kwargs...) = f(x, sum(args); kwargs...)

f(1, 2, 3, 4, 5, z = 6)  # 21
```